internal/dag/policy.go

```diff
@@ -676,17 +676,28 @@ func loadBalancerRequestHashPolicies(lbp *contour_api_v1.LoadBalancerPolicy, val
 		actualStrategy := strategy
 		hashSourceIPSet := false
 		// Map of unique header names.
-		headerHashPolicies := map[string]bool{}
+		headerHashPolicies := map[string]struct{}{}
```

Comment:
Definitely not required to change now but `sets.String` is a nice little API for this kind of thing, see e.g. https://github.com/projectcontour/contour/blob/main/internal/dag/gatewayapi_processor.go#L535-L568 for usage.